### DISCUSSION TITLE
HomeSpan examples

### DISCUSSION THREAD

**Discussion by alexup66** [HomeSpan/HomeSpan#610] [General]
Firstly, thank you, Gregg E. Berman, for this amazing HomeSpan project! I have done some sketches with 2 switches/relays controlled by 2 spanbuttons, 1 dimmable LED controlled by a spanbutton, and a fading LED controlled by a spanbutton using your examples. Everything works okay, but I have a question: Sometimes not all the status updates are reflected in the Home apps. I have to close and reopen the apps to see the updated status.

**Comment by HomeSpan** [comment]
Thanks!  Really glad to hear you are enjoying the library.

Re the status updates, is the problem that when you press the button to control the light locally, it does not get reflected in the Home App (even though you are using setVal() to send the notification)?   Apple seems to have made a lot of changes in the backend of HomeKit earlier this year and I had the same issue.   It would take as long as 30 seconds for the Home App to reflect a setVal() update.  However, after updating the Home App to use the new HomeKit architecture the issue resolved itself (notifications are now reflected in the HomeApp within less than a second).

Have you upgraded to the new HomeKit architecture?   Also, are you using a HomeKit Hub, such as a HomePod or Apple TV?

**Comment by alexup66** [comment]
Thank you for the prompt response. Yes, I have tried using setVal() as well. I am currently using the iOS 17 beta. In the Home app, I couldn't find a way to upgrade to a new HomeKit architecture. Moreover, I am not using any HomeKit Hub and have a direct connection to my home network on the 2.4GHz band. However, I'm facing an issue where I have to close and reopen apps every time to get a status update. Could this problem be related to the iOS 17 beta version?

**Comment by HomeSpan** [comment]
This problem is very similar to the one in discussion #594, though in that case I believe the Accessory stopped responding completely. 

If I'm understanding correctly, the issue you are having is that you can operate the device properly from the Home App, and if you turn on the light from the Home App, the Home App icon will correctly show it is on, BUT if you instead turn the light on via a button, the status is not properly reflected in the Home App.

If so, this is caused by lack of a HomeKit Hub.  My guess is that iOS 17 beta has the new HomeKit architecture built in, and that architecture (which is replacing the old one) requires the use of a Home Hub to receive any notifications.  I have many devices and use a HomePod as my Home Hub, but to confirm this is the problem, I created a new "Test Home" in the Home App and added just a single device, without any Hubs.  In this configuration I was able to operate the device from the Home App, but any changes made locally with a SpanButton, as well as updates from a temperature sensor, were NOT reflected in the Home App.  Closing and restarting the Home App was the only way to get the Home App to properly reflect the new status.  This is because when the Home App first starts, it asks each device for a full dump of all Characteristics, so it retrieves the latest status.

Unfortunately, the only way to solve this is by adding a HomePod or Apple TV as a Home Hub.  Apple made this change towards the end of last year.  See https://support.apple.com/en-us/HT213481 and related links for details.

**Comment by alexup66** [comment]
Now, everything is clear. Thank you very much for your quick response, Gregg.

**Comment by alexup66** [comment]
Dear Greg,Need your help,Could you please answer? I have HomeKit-compatible MFi devices installed in my home apps. Additionally, I’m using Apple TV. Everything is working correctly. I’m trying to add HomeSpan devices without success. When pairing the device, I see it just one time and can’t add it to my home. (Tried several times, several sketches without sucsess). My question is, can MFi and HomeSpan devices work in the same home?” This work for me if i totally removed all home with MFI devices, and then is possible add all homespan devices.Thank you!Regards,Alex.Sent from my iPhoneOn 21 Jul 2023, at 6:34, HomeSpan ***@***.***> wrote:﻿
This problem is very similar to the one in discussion #594, though in that case I believe the Accessory stopped responding completely.
If I'm understanding correctly, the issue you are having is that you can operate the device properly from the Home App, and if you turn on the light from the Home App, the Home App icon will correctly show it is on, BUT if you instead turn the light on via a button, the status is not properly reflected in the Home App.
If so, this is caused by lack of a HomeKit Hub.  My guess is that iOS 17 beta has the new HomeKit architecture built in, and that architecture (which is replacing the old one) requires the use of a Home Hub to receive any notifications.  I have many devices and use a HomePod as my Home Hub, but to confirm this is the problem, I created a new "Test Home" in the Home App and added just a single device, without any Hubs.  In this configuration I was able to operate the device from the Home App, but any changes made locally with a SpanButton, as well as updates from a temperature sensor, were NOT reflected in the Home App.  Closing and restarting the Home App was the only way to get the Home App to properly reflect the new status.  This is because when the Home App first starts, it asks each device for a full dump of all Characteristics, so it retrieves the latest status.
Unfortunately, the only way to solve this is by adding a HomePod or Apple TV as a Home Hub.  Apple made this change towards the end of last year.  See https://support.apple.com/en-us/HT213481 and related links for details.

**Comment by HomeSpan** [comment]
You can mix and max all HomeKit products --- there is no problem with using HomeSpan alongside of commercial products.

Note that if you can't see a HomeSpan device when you try to pair, that likely means it was previously paired already and you did not unpair it.  To solve this, simply enter 'H' into the Serial Monitor.  This causes HomeSpan to delete all pairing information and reboots with a new device ID ready for pairing.
